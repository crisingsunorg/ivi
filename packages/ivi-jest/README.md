# Jest utils for ivi library
