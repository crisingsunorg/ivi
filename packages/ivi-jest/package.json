{
  "private": true,
  "name": "ivi-jest",
  "version": "0.0.1",
  "main": "dist/main/index.js",
  "module": "dist/module/index.js",
  "sideEffects": false,
  "files": [
    "dist",
    "src",
    "!dist/**/*.tsbuildinfo",
    "!src/**/__tests__",
    "README.md",
    "LICENSE"
  ],
  "scripts": {
    "prepublishOnly": "npm run clean && npm run dist",
    "clean": "rm -rf ./dist",
    "dist": "npm run dist:main && npm run dist:module",
    "dist:main": "../../node_modules/.bin/tsc -p ./tsconfig.build.main.json",
    "dist:module": "../../node_modules/.bin/tsc -p ./tsconfig.build.module.json"
  },
  "dependencies": {
    "jsdom": "^14.0.0"
  },
  "peerDependencies": {},
  "devDependencies": {},
  "description": "Jest utils for ivi library",
  "license": "MIT",
  "author": {
    "name": "Boris Kaul",
    "email": "localvoid@gmail.com",
    "url": "https://github.com/localvoid"
  },
  "keywords": [
    "ivi"
  ],
  "homepage": "https://github.com/localvoid/ivi",
  "repository": {
    "type": "git",
    "url": "https://github.com/localvoid/ivi"
  },
  "bugs": {
    "url": "https://github.com/localvoid/ivi/issues"
  },
  "github": "https://github.com/localvoid/ivi"
}
