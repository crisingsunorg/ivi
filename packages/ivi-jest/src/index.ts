export { useMockFn, useSpyOn } from "./mock";
export { useResetModules, useModule } from "./module";
export { useDOMElement, useDOMOpsCounters } from "./dom";
