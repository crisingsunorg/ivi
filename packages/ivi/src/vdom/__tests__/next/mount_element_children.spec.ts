import { useResetModules, useModule, useDOMElement, useDOMOpsCounters } from "ivi-jest";

useResetModules();
const c = useDOMElement();
const domOps = useDOMOpsCounters();
const h = useModule<typeof import("ivi-html")>("ivi-html");
const t = useModule<typeof import("ivi-test")>("ivi-test");
const _ = void 0;

describe("mount element children", () => {
  test("null child", () => {
    const { domNode } = t.render(h.div(_, _, null), c());
    expect(domNode).toMatchSnapshot();
    expect(domOps()).toMatchSnapshot();
  });

  test("single child", () => {
    const { domNode } = t.render(h.div(_, _, "abc"), c());
    expect(domNode).toMatchSnapshot();
    expect(domOps()).toMatchSnapshot();
  });
});
