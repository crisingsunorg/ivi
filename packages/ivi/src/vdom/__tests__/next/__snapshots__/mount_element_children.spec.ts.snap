// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount element children null child 1`] = `<div />`;

exports[`mount element children null child 2`] = `
Object {
  "appendChild": 0,
  "createElement": 1,
  "createElementNS": 0,
  "createTextNode": 0,
  "innerHTML": 0,
  "insertBefore": 1,
  "nodeValue": 0,
  "removeChild": 0,
  "replaceChild": 0,
  "textContent": 0,
}
`;

exports[`mount element children single child 1`] = `
<div>
  abc
</div>
`;

exports[`mount element children single child 2`] = `
Object {
  "appendChild": 0,
  "createElement": 1,
  "createElementNS": 0,
  "createTextNode": 1,
  "innerHTML": 0,
  "insertBefore": 2,
  "nodeValue": 0,
  "removeChild": 0,
  "replaceChild": 0,
  "textContent": 0,
}
`;
